viewer/src/main/java/rv/Configuration.java

```diff
@@ -263,79 +469,147 @@ private void write(BufferedWriter out) throws IOException
-		return (line.length() > 0 && !line.startsWith("#") &&
-				!(line.equals("Graphics Settings:") || line.equals("Overlay Default Visibility:") ||
-						line.equals("Networking Settings:") || line.equals("General Settings:") ||
-						line.equals("Team Colors:")));
+		return (line.trim().length() > 0 && !line.startsWith("#"));
 	}
 
-	private void parseLine(String line)
+	private static Pair<String, String> parseLine(String line)
 	{
-		String key = line.substring(0, line.indexOf(":") - 1).trim();
-		String val = line.substring(line.indexOf(":") + 1).trim();
-
-		configList.add(new Pair<>(key, val));
+		try {
+			String key = line.substring(0, line.indexOf(":")).trim();
+			String val = line.substring(line.indexOf(":") + 1).trim();
+			return new Pair<>(key, val);
+		} catch (IndexOutOfBoundsException e) {
+			// Line doesn't contain a colon
+			System.err.println("\"" + line + "\" is not a valid key-value pair.");
+			return null;
+		}
 	}
 
-	private static String getValue(String key)
+	private String getValue(String key)
 	{
 		Pair<String, String> valuePair = configList.stream().filter(it -> it.getFirst().equals(key)).findFirst().get();
 
-		return valuePair.getSecond();
+		if (valuePair != null) {
+			return valuePair.getSecond();
+		} else {
+			return "";
+		}
+	}
+
+	private List<String> getAllValues(String key)
+	{
+		return configList.stream()
+				.filter(it -> it.getFirst().equals(key))
+				.map(it -> it.getSecond())
+				.collect(Collectors.toList());
 	}
 
-	private static boolean getBool(String key)
+	private boolean getBool(String key)
 	{
 		return Boolean.parseBoolean(getValue(key));
 	}
 
-	private static int getInt(String key)
+	private int getInt(String key)
 	{
 		return Integer.parseInt(getValue(key));
 	}
 
-	private static Integer getInteger(String key)
+	private String getString(String key)
 	{
+		return getValue(key);
+	}
+
+	private static Pair<String, String> decodeValueValuePair(String pair)
+	{
+		String val1 = "";
+		String val2 = "";
 		try {
-			return Integer.parseInt(getValue(key));
-		} catch (NumberFormatException e) {
+			val1 = pair.substring(0, pair.indexOf(",")).trim();
+			val2 = pair.substring(pair.indexOf(",") + 1).trim();
+		} catch (IndexOutOfBoundsException e) {
+			// Line doesn't contain a comma
+			System.err.println("\"" + pair + "\" is not a valid csv-value pair.");
 			return null;
 		}
-	}
 
-	private static String getString(String key)
-	{
-		return getValue(key);
+		return new Pair<>(val1, val2);
 	}
 
-	private static String[] getStringList(String key)
+	private static String encodeValueValuePair(String val1, String val2)
 	{
-		return getValue(key).split(", ?");
+		return val1 + "," + val2;
 	}
 
-	private static void writeSection(Writer out, String name) throws IOException
+	private static void writeConfigLine(List<String> configLines, String name, String newString)
 	{
-		out.write(name + ":" + getNewline());
+		// Escape special regex characters
+		name = name.replace("\\", "\\\\");
```

Comment:
These lines seem to be duplicated here and in `writeValList()`, should probably be extracted into a common method.